python/cuml/manifold/t_sne.pyx

```diff
@@ -0,0 +1,472 @@
+            print("[Error] angle = {} should be > 0 and less "
+                  "than 1.".format(angle))
+            angle = 0.5
+        if n_neighbors < 0:
+            print("[Error] n_neighbors = {} should be more "
+                  "than 0.".format(n_neighbors))
+            n_neighbors = <int> (perplexity * 3)
+        if n_neighbors > 1023:
+            print("[Error] n_neighbors = {} should be less than 1023, as "
+                  "FAISS doesn't support more".format(n_neighbors))
+            n_neighbors = 1023
+        if perplexity_max_iter < 0:
+            print("[Error] perplexity_max_iter = {} should be more "
+                  "than 0.".format(perplexity_max_iter))
+            perplexity_max_iter = 100
+        if exaggeration_iter < 0:
+            print("[Error] exaggeration_iter = {} should be more "
+                  "than 0.".format(exaggeration_iter))
+            exaggeration_iter = 250
+        if exaggeration_iter > n_iter:
+            print("[Error] exaggeration_iter = {} should be more less than "
+                  "n_iter = {}.".format(exaggeration_iter, n_iter))
+            exaggeration_iter = <int> max(<float>n_iter * 0.25, 1)
+        if pre_momentum < 0 or pre_momentum > 1:
+            print("[Error] pre_momentum = {} should be more than 0 and less "
+                  "than 1.".format(pre_momentum))
+            pre_momentum = 0.5
+        if post_momentum < 0 or post_momentum > 1:
+            print("[Error] post_momentum = {} should be more than 0 and less "
+                  "than 1.".format(post_momentum))
+            post_momentum = 0.8
+        if pre_momentum > post_momentum:
+            print("[Error] post_momentum = {} should be more than "
+                  "pre_momentum = {}".format(post_momentum, pre_momentum))
+            pre_momentum = post_momentum * 0.75
+
+        self.n_components = n_components
+        self.perplexity = perplexity
+        self.early_exaggeration = early_exaggeration
+        self.learning_rate = learning_rate
+        self.n_iter = n_iter
+        self.n_iter_without_progress = n_iter_without_progress
+        self.min_grad_norm = min_grad_norm
+        self.metric = metric
+        self.init = init
+        self.verbose = verbose
+        self.random_state = random_state
+        self.method = method
+        self.angle = angle
+        self.n_neighbors = n_neighbors
+        self.perplexity_max_iter = perplexity_max_iter
+        self.exaggeration_iter = exaggeration_iter
+        self.pre_momentum = pre_momentum
+        self.post_momentum = post_momentum
+        self.learning_rate_method = learning_rate_method
+
+        self.epssq = 0.0025
+        self.perplexity_tol = 1e-5
+        self.min_gain = 0.01
+        self.pre_learning_rate = learning_rate
+        self.post_learning_rate = learning_rate * 2
+
+        self._should_downcast = should_downcast
+        self.Y = None
+        return
+
+    def __repr__(self):
```

Comment:
Your PR https://github.com/rapidsai/cuml/pull/824 Adds this to base, so no need to have this here